src/LuaInterface.cpp

```diff
@@ -342,43 +342,44 @@ bool LuaInterface::setCValue(QList<TVar*> vars)
     return false;
 }
 
+// sets the value of a Lua variable by running dynamically-generated Lua code
 bool LuaInterface::setValue(TVar* var)
 {
     //This function assumes the var has been modified and then called
     L = interpreter->pGlobalLua;
 
     QList<TVar*> vars = varOrder(var);
-    QString newName = vars[0]->getName();
+    QString variableChangeCode = vars[0]->getName();
     for (int i = 1; i < vars.size(); i++) {
         if (vars[i]->isReference()) {
             return setCValue(vars);
         }
         if (vars[i]->getKeyType() == LUA_TNUMBER) {
-            newName.append("[" + vars[i]->getName() + "]");
+            variableChangeCode.append("[" + vars[i]->getName() + "]");
```

Comment:
Let's avoid raw literals when dealing with `QString`s, so we do not have to revisit when `QT_NO_CAST_FROM_ASCII` and `QT_NO_CAST_TO_ASCII` get defined (and use faster `const QList<T>::at()` instead of `QList<T>::operator[]` accessor to `vars`, which should be used where possible):

    variableChangeCode.append(QStringLiteral("[%1]").arg(vars.at(i)->getName());